src/jit/gentree.h

```diff
@@ -992,6 +994,55 @@ struct GenTree
         return opKind & ~GTK_EXOP;
     }
 
+    bool IsValue() const
+    {
+        if ((OperKind(gtOper) & GTK_NOVALUE) != 0)
+        {
+            return false;
+        }
+
+        if (gtOper == GT_NOP || gtOper == GT_CALL)
+        {
+            return gtType != TYP_VOID;
+        }
+
+        return true;
+    }
+
+    bool IsLIR() const
+    {
+        if ((OperKind(gtOper) & GTK_NOTLIR) != 0)
+        {
+            return false;
+        }
+
+        switch (gtOper)
+        {
+            case GT_NOP:
+                // NOPs may only be present in LIR if they do not produce a value.
+                return IsNothingNode();
+
+            case GT_ARGPLACE:
+            case GT_LIST:
+                // ARGPLACE and LIST nodes may not be present in a block's LIR sequence, but they may
+                // be used to represent non-executable concepts.
```

Comment:
> I believe that you are saying that these nodes may be present as children of an LIR node, but, since they do not result in executable code, they are not present in the code sequence. 

This is correct. I'll rephrase the comment.